examples/ensemble/plot_gradient_boosting_categorical.py

```diff
@@ -272,6 +266,5 @@ def plot_results(figure_title):
 # %%
 # The results for these under-fitting models confirm our previous intuition:
 # the native category handling strategy performs the best when the splitting
-# budget is constrained. The two other strategies (one-hot encoding and
-# treating categories as ordinal values) lead to error values comparable
-# to the baseline model that just dropped the categorical features altogether.
```

Comment:
This remark was no longer true since the example was made to run on a subset of the columns to make it run faster. Compare the last plot of:

- https://scikit-learn.org/0.24/auto_examples/ensemble/plot_gradient_boosting_categorical.html#limitting-the-number-of-splits

and:

- https://scikit-learn.org/1.0/auto_examples/ensemble/plot_gradient_boosting_categorical.html#limitting-the-number-of-splits

So I fixed this as part of the PR (along with a simplification of the column indexing code.